File: data structure in c/linearSearch.c
#include<stdio.h>

int linearSearch(int arr[] , int n , int key ){
    for (int i = 0; i < n; i++){
        if (arr[i] == key){
            return i;
        }
    }
    return -1;
}

void main(){
    int x;
    printf("Enter length of array: ");
    scanf("%d", &x);
    int arr[x];
    printf("Enter array elements : \n");
    for (int i = 0; i < x; i++){
        scanf("%d", &arr[i]);
    }
    int tar;
    printf("Enter target element to search :");
    scanf("%d", &tar);
    int result = linearSearch(arr , x , tar);
    if (result != -1){
        printf("Target found at location/index : %d\n", result);
    }
    else{
        printf("Target not found exit code %d\n", result);
        
    }
}

File: python/linearSearch.py
def linearSearch(arr, key):
    for i in range(len(arr)):
        if arr[i] == key:
            return i
    return -1


array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
key = int(input("Enter key between 1 to 9 :"))
index = linearSearch(array, key)

if index == -1:
    print("key", key, "is not found")
else:
    print("key", key, "is found at index", index)
